src/main/kotlin/com/exactpro/th2/common/utils/event/MessageUtils.kt

```
@@ -1,23 +1,34 @@
-fun RawMessage.toGroup(): MessageGroup = MessageGroup.newBuilder().addMessage(this).build()
+fun Message.toGroup(): MessageGroup = MessageGroup.newBuilder().add(this).build()
+fun RawMessage.toGroup(): MessageGroup = MessageGroup.newBuilder().add(this).build()
+fun AnyMessage.toGroup(): MessageGroup = MessageGroup.newBuilder().addMessages(this).build()
 
-fun MessageGroup.Builder.addMessage(message: Message): MessageGroup.Builder = this.addMessages(AnyMessage.newBuilder().setMessage(message).build())
-fun MessageGroup.Builder.addMessage(message: RawMessage): MessageGroup.Builder = this.addMessages(AnyMessage.newBuilder().setRawMessage(message).build())
+val AnyMessageOrBuilder.sessionAlias: String
+    get() = when {
+        hasMessage() -> message.metadata.id.connectionId.sessionAlias
+        hasRawMessage() -> message.metadata.id.connectionId.sessionAlias
+        else -> error("Unsupported message kind: $kindCase")
+    }
+
+fun MessageGroup.Builder.add(message: Message): MessageGroup.Builder = apply { addMessagesBuilder().message = message }
+fun MessageGroup.Builder.add(message: RawMessage): MessageGroup.Builder = apply { addMessagesBuilder().rawMessage = message }
 
-fun MessageGroupOrBuilder.getSessionAlias(): String? {
-    messagesList.forEach {
-        when {
-            it.hasMessage() && !it.message.metadata.id.connectionId.sessionAlias.isNullOrBlank() -> return it.message.metadata.id.connectionId.sessionAlias
-            it.hasRawMessage() && !it.message.metadata.id.connectionId.sessionAlias.isNullOrBlank() -> return it.message.metadata.id.connectionId.sessionAlias
+val MessageGroupOrBuilder.sessionAlias: String?
+    get() {
+        messagesList.forEach {
+            with(it.sessionAlias) {
+                if (isNotBlank()) {
+                    return this
+                }
+            }
         }
-    }
-    return null
-}
\ No newline at end of file
+        return null
```

Let's throw an exception if there's more than one session alias or none at all